typegraph/deno/src/types.ts

```diff
@@ -56,6 +64,13 @@ export class Typedef {
   asTypedef(): Typedef {
     return new Typedef(this._id, { name: this.name });
   }
+
+  optional(data: Simplified<TypeOptional> = {}): Optional {
```

Comment:
this may cause issue for struct having a field called optional, I wonder whether we may avoid this syntax for now?

Reply:
Definitely, that could occur when class syntax is used,  I think we can change it to `optional_()` or `asOptional()`.